components/clarity-repl/src/repl/remote_data/context.rs

```diff
@@ -0,0 +1,100 @@
+use clarity::vm::contexts::GlobalContext;
+use clarity::vm::database::{
+    DataMapMetadata, DataVariableMetadata, FungibleTokenMetadata, NonFungibleTokenMetadata,
+};
+use clarity::vm::errors::InterpreterResult as Result;
+use clarity::vm::functions::define::DefineResult;
+use clarity::vm::types::PrincipalData;
+use clarity::vm::{
+    eval, functions, CallStack, ContractContext, Environment, LocalContext, SymbolicExpression,
+    Value,
+};
+
+#[allow(clippy::result_large_err)]
+pub fn set_contract_context(
+    expressions: &[SymbolicExpression],
+    contract_context: &mut ContractContext,
+    global_context: &mut GlobalContext,
+) -> Result<Option<Value>> {
+    let mut last_executed = None;
+    let context = LocalContext::new();
+
+    // todo: should set contract_context.data_size?
+
+    let publisher: PrincipalData = contract_context.contract_identifier.issuer.clone().into();
+
+    for exp in expressions {
+        let try_define = global_context.execute(|g| {
+            let mut call_stack = CallStack::new();
+            let mut env = Environment::new(
+                g,
+                contract_context,
+                &mut call_stack,
+                Some(publisher.clone()),
+                Some(publisher.clone()),
+                None,
+            );
+            functions::define::evaluate_define(exp, &mut env)
```

Comment:
Do you actually need to evaluate these, or can you just build the context just based on the expressions? None of the names or types would be dependent on evaluation the expressions, only the values, which I think we shouldn't care about, since we'd still want to get those from the actual contract.

Reply:
If we can make sure that these constants get pulled from the mxs and not from here, then we could ignore this value.

Reply:
Ok I see the point.
Well then I'm pretty confident that the data doesn't get the data from there inded.
I'll try to simplify to just the types then. 

What should I insert in `contract_context.variables`? Empty values?

Reply:
Yeah, whatever is simplest to create.